src/java.desktop/share/classes/javax/swing/text/html/HTMLDocument.java

```diff
@@ -3423,47 +3423,53 @@ public void start(HTML.Tag t, MutableAttributeSet attr) {
                 if (styleAttributes != null) {
                     charAttr.addAttributes(styleAttributes);
                 }
+
+                convertAttributes(t, attr);
             }
 
             public void end(HTML.Tag t) {
                 popCharacterStyle();
             }
+
+            /**
+             * Converts HTML tags to CSS attributes.
+             * @param t the current HTML tag
+             * @param attr the attributes of the HTML tag
+             */
+            void convertAttributes(HTML.Tag t, MutableAttributeSet attr) {
+            }
+        }
+
+        final class ConvertSpanAction extends CharacterAction {
+            @Override
+            void convertAttributes(HTML.Tag t, MutableAttributeSet attr) {
+                Object newDecoration = attr.getAttribute(CSS.Attribute.TEXT_DECORATION);
```

Comment:
parameter 't' isn't used. Why is that ?

Reply:
Because `ConvertSpanAction` is used for `HTML.Tag.SPAN` only:

https://github.com/openjdk/jdk/blob/e9cf30b3b31b50db8371bb7fdbfc2d68aacbb7da/src/java.desktop/share/classes/javax/swing/text/html/HTMLDocument.java#L2502